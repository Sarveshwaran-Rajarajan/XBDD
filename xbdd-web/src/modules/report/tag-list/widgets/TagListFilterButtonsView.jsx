import React from "react";
import { PropTypes } from "prop-types";
import { withStyles, MuiThemeProvider, createMuiTheme } from "@material-ui/core/styles";
import IconButton from "@material-ui/core/IconButton";
import Tooltip from "@material-ui/core/Tooltip";
import DoneIcon from "@material-ui/icons/Done";
import ErrorOutlineIcon from "@material-ui/icons/ErrorOutline";
import BlockIcon from "@material-ui/icons/Block";
import HelpOutlineIcon from "@material-ui/icons/HelpOutline";
import { tagListFilterButtonStyles } from "../styles/TagListStyles";
import { Card } from "@material-ui/core";

const buttonTheme = color =>
  createMuiTheme({
    palette: {
      primary: { main: color },
      secondary: { main: "#E0E0E0" },
    },
    typography: {
      useNextVariants: true,
    },
  });

const createButton = variant => (
  <MuiThemeProvider theme={variant.theme}>
    <Tooltip title={variant.tooltip} placement="top">
      <IconButton className={variant.className} onClick={variant.handler} variant="outlined" size="small" color={variant.color}>
        {variant.icon}
      </IconButton>
    </Tooltip>
  </MuiThemeProvider>
);

const TagListFilterButtonsView = props => {
  const { classes, filterStates, onFilterButtonClick } = props;
  const variants = {
    passed: {
      theme: buttonTheme("#1d3557"),
      icon: <DoneIcon className={classes.xbddFilterButtonIcon} />,
      tooltip: "Passed",
      color: filterStates.passedSelected ? "primary" : "secondary",
      handler: () => onFilterButtonClick("passedSelected"),
      className: `${classes.xbddFilterButton} ${classes.xbddFilterButtonFirst}`,
    },
    failed: {
      theme: buttonTheme("#E63946"),
      icon: <ErrorOutlineIcon className={classes.xbddFilterButtonIcon} />,
      tooltip: "Failed",
      color: filterStates.failedSelected ? "primary" : "secondary",
      handler: () => onFilterButtonClick("failedSelected"),
      className: `${classes.xbddFilterButton}`,
    },
    undefined: {
      theme: buttonTheme("#A8DADC"),
      icon: <HelpOutlineIcon className={classes.xbddFilterButtonIcon} />,
      tooltip: "Undefined",
      color: filterStates.undefinedSelected ? "primary" : "secondary",
      handler: () => onFilterButtonClick("undefinedSelected"),
      className: `${classes.xbddFilterButton}`,
    },
    skipped: {
      theme: buttonTheme("#457B9D"),
      icon: <BlockIcon className={classes.xbddFilterButtonIcon} />,
      tooltip: "Skipped",
      color: filterStates.skippedSelected ? "primary" : "secondary",
      handler: () => onFilterButtonClick("skippedSelected"),
      className: `${classes.xbddFilterButton} ${classes.xbddFilterButtonLast}`,
    },
  };

  return (
    <Card raised className={classes.xbddTagListFilterButtons}>
      {createButton(variants.passed)}
      {createButton(variants.failed)}
      {createButton(variants.undefined)}
      {createButton(variants.skipped)}
    </Card>
  );
};

TagListFilterButtonsView.propTypes = {
  filterStates: PropTypes.shape({}).isRequired,
  onFilterButtonClick: PropTypes.func.isRequired,
  classes: PropTypes.shape({}).isRequired,
};

export default withStyles(tagListFilterButtonStyles)(TagListFilterButtonsView);
