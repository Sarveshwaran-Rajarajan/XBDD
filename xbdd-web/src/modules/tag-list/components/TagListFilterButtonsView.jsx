import React from 'react';
import { PropTypes } from 'prop-types';
import { withStyles, MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import Tooltip from '@material-ui/core/Tooltip';
import DoneIcon from '@material-ui/icons/Done';
import ErrorOutlineIcon from '@material-ui/icons/ErrorOutline';
import BlockIcon from '@material-ui/icons/Block';
import HelpOutlineIcon from '@material-ui/icons/HelpOutline';
import { tagListFilterButtonStyles } from '../styles/TagListStyles';

const buttonTheme = color => createMuiTheme({
    palette: {
        primary: { main: color },
        secondary: { main: '#E0E0E0' },
    },
});

const createButton = variant => (
    <MuiThemeProvider theme={variant.theme}>
        <Tooltip title={variant.tooltip} placement="top">
            <IconButton
                className={variant.className}
                onClick={variant.handler}
                variant="outlined"
                size="small"
                color={variant.color}
            >
                {variant.icon}
            </IconButton>
        </Tooltip>
    </MuiThemeProvider>
);

const TagListFilterButtonsView = (props) => {
    const variants = {
        passed: {
            theme: buttonTheme('#1d3557'),
            icon: <DoneIcon />,
            tooltip: 'Passed',
            color: props.filterStates.passedSelected ? 'primary' : 'secondary',
            handler: () => props.onFilterButtonClick('passedSelected'),
            className: `${props.classes.xbddFilterButton} ${props.classes.xbddFilterButtonFirst}`,
        },
        failed: {
            theme: buttonTheme('#E63946'),
            icon: <ErrorOutlineIcon />,
            tooltip: 'Failed',
            color: props.filterStates.failedSelected ? 'primary' : 'secondary',
            handler: () => props.onFilterButtonClick('failedSelected'),
            className: `${props.classes.xbddFilterButton}`,
        },
        undefined: {
            theme: buttonTheme('#A8DADC'),
            icon: <HelpOutlineIcon />,
            tooltip: 'Undefined',
            color: props.filterStates.undefinedSelected ? 'primary' : 'secondary',
            handler: () => props.onFilterButtonClick('undefinedSelected'),
            className: `${props.classes.xbddFilterButton}`,
        },
        skipped: {
            theme: buttonTheme('#457B9D'),
            icon: <BlockIcon />,
            tooltip: 'Skipped',
            color: props.filterStates.skippedSelected ? 'primary' : 'secondary',
            handler: () => props.onFilterButtonClick('skippedSelected'),
            className: `${props.classes.xbddFilterButton} ${props.classes.xbddFilterButtonLast}`,
        },
    };

    return (
        <div className={props.classes.xbddTagListFilterButtons}>
            {createButton(variants.passed)}
            {createButton(variants.failed)}
            {createButton(variants.undefined)}
            {createButton(variants.skipped)}
        </div>
    );
};

TagListFilterButtonsView.propTypes = {
    filterStates: PropTypes.object.isRequired,
    onFilterButtonClick: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired,
};

export default withStyles(tagListFilterButtonStyles)(TagListFilterButtonsView);
