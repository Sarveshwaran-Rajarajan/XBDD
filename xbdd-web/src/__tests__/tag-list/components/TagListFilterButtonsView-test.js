import React from "react";
import { shallow, mount } from "enzyme";
import TagListFilterButtonsView from "../../../modules/tag-list/widgets/TagListFilterButtonsView";

const filterStates = {
  passedSelected: true,
  undefinedSelected: true,
  failedSelected: true,
  skippedSelected: true,
};

describe("TagListFilterButtonView", () => {
  test("renders", () => {
    const buttonView = shallow(<TagListFilterButtonsView filterStates={filterStates} onFilterButtonClick={() => {}} />);
    expect(buttonView).toMatchSnapshot();
  });

  test("renders 4 buttons", () => {
    const buttonView = mount(<TagListFilterButtonsView filterStates={filterStates} onFilterButtonClick={() => {}} />);
    expect(buttonView.find("button")).toHaveLength(4);
  });

  test("renders button with primary color when selected", () => {
    const buttonView = mount(<TagListFilterButtonsView filterStates={filterStates} onFilterButtonClick={() => {}} />);
    expect(buttonView.find("button").get(0).props.className).toContain("Primary");
  });

  test("renders button with secondary color when not selected", () => {
    const filterStatesFalse = {
      passedSelected: false,
      undefinedSelected: false,
      failedSelected: false,
      skippedSelected: false,
    };
    const buttonView = mount(<TagListFilterButtonsView filterStates={filterStatesFalse} onFilterButtonClick={() => {}} />);
    expect(buttonView.find("button").get(0).props.className).toContain("Secondary");
  });
});
