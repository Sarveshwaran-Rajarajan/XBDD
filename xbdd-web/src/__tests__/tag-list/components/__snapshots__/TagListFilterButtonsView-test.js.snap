// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagListFilterButtonView renders 1`] = `ShallowWrapper {}`;
